src/main/java/bisq/core/dao/bonding/lockup/LockupService.java

```diff
@@ -0,0 +1,145 @@
+
+    ///////////////////////////////////////////////////////////////////////////////////////////
+    // Constructor
+    ///////////////////////////////////////////////////////////////////////////////////////////
+
+    @Inject
+    public LockupService(WalletsManager walletsManager,
+                         BsqWalletService bsqWalletService,
+                         BtcWalletService btcWalletService,
+                         StateService stateService) {
+        this.walletsManager = walletsManager;
+        this.bsqWalletService = bsqWalletService;
+        this.btcWalletService = btcWalletService;
+        this.stateService = stateService;
+    }
+
+    public void publishLockupTx(Coin lockupAmount, int lockupTime, ResultHandler resultHandler,
+                                ExceptionHandler exceptionHandler) {
+        checkArgument(lockupTime <= stateService.getParamValue(Param.LOCKTIME_MAX, stateService.getChainHeight()) &&
+                lockupTime >= stateService.getParamValue(Param.LOCKTIME_MIN, stateService.getChainHeight()));
+        try {
+            byte[] opReturnData = getOpReturnData(lockupTime);
+            final Transaction lockupTx = getLockupTx(lockupAmount, opReturnData);
+
+            walletsManager.publishAndCommitBsqTx(lockupTx, new TxBroadcaster.Callback() {
+                @Override
+                public void onSuccess(Transaction transaction) {
+                    resultHandler.handleResult();
+                }
+
+                @Override
+                public void onTimeout(TxBroadcastTimeoutException exception) {
+                    exceptionHandler.handleException(exception);
+                }
+
+                @Override
+                public void onTxMalleability(TxMalleabilityException exception) {
+                    exceptionHandler.handleException(exception);
+                }
+
+                @Override
+                public void onFailure(TxBroadcastException exception) {
+                    exceptionHandler.handleException(exception);
+                }
+            });
+
+        } catch (TransactionVerificationException | InsufficientMoneyException | WalletException |
+                IOException exception) {
+            exceptionHandler.handleException(exception);
+        }
+    }
+
+    private byte[] getOpReturnData(int lockupTime) throws IOException {
+        try (ByteArrayOutputStream outputStream = new ByteArrayOutputStream()) {
+            outputStream.write(OpReturnType.LOCKUP.getType());
+            outputStream.write(Version.LOCKUP_VERSION);
+            outputStream.write(lockupTime >>> 8);
+            outputStream.write(lockupTime);
+            // TODO: handle short data
+            // Pushdata of <= 4 bytes is converted to int when returned from bitcoind and not handled the way we
+            // require by btcd-cli4j
+            // Write an extra byte to avoid the asm conversion to int in bitcoind
+            outputStream.write(0);
```

Comment:
I think we should add a subType anyway to have more flexibility for diff. types of locks